### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by jmmv**
The various changes in this PR upgrade all components of the `WORKSPACE` to newer versions and also make our Travis CI builds use a modern Bazel. These changes are all required to allow sandboxfs to use recent Bazel versions, as the old Go rules we used stopped working.

### CODE REVIEW

File: admin/lint/checks.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 
 // checkLicense checks if the given file contains the necessary license information and returns an
 // error if this is not true or if the check cannot be performed.
-func checkLicense(file string) error {
+func checkLicense(workspaceDir string, file string) error {
```
Comment (jin) [inline admin/lint/checks.go]: unrelated changes?

Comment (jmmv) [reply admin/lint/checks.go]: No, these were necessary. The workspace directory changes were already in a separate commit with a justification for them (to fix Gazelle runs) and I have now also split out the test changes to explain why they are necessary (to pass with Go 1.11).
